src/redprl/redprl.grm

```diff
@@ -86,16 +87,44 @@ struct
   val makeDFun = makeQuant O.DFUN o flatten
   val makeDProd = makeQuant O.DPROD o flatten
 
-  fun makeRecord tbs =
+  structure LabelSet = SplaySet (structure Elem = StringOrdered)
+
+
+  fun checkNoDuplicates pos lbls = 
     let
-      val (lbls, tys) = ListPair.unzip (flatten tbs)
+      fun insert (lbl, set) = 
+        if LabelSet.member set lbl then
+          raise RedPrlError.annotate pos (RedPrlError.error [Fpp.text "Duplicate label", Fpp.text lbl])
+        else
+          LabelSet.insert set lbl
```

Comment:
This can be done by `operate'` (or `operate`?) in one step, but maybe you don't want to use clever tricks...